Extensions/TokenAcquisition.cs

```diff
@@ -159,35 +172,53 @@ public async Task AddAccountToCacheFromAuthorizationCode(AuthorizationCodeReceiv
         /// <param name="user">Account described by its claims</param>
         /// <param name="scopes">Scopes to request for the downstream API to call</param>
         /// <returns>An access token to call the downstream API characterized by its scopes, on behalf of the user</returns>
-        public async Task<string> GetAccessTokenOnBehalfOfUser(HttpContext context, ClaimsPrincipal user, string[] scopes)
+        public async Task<string> GetAccessTokenOnBehalfOfUser(HttpContext context, string[] scopes)
         {
             if (context == null)
                 throw new ArgumentNullException(nameof(context));
 
-            if (user == null)
-                throw new ArgumentNullException(nameof(user));
-
             if (scopes == null)
                 throw new ArgumentNullException(nameof(scopes));
 
             // Use MSAL to get the right token to call the API
-            var credential = new ClientCredential(_azureAdOptions.ClientSecret);
-            var request = context.Request;
-            var currentUri = UriHelper.BuildAbsolute(request.Scheme, request.Host, request.PathBase, request.Path);
-            Application = new ConfidentialClientApplication(_azureAdOptions.ClientId, currentUri, new ClientCredential(_azureAdOptions.ClientSecret), 
-                AuthPropertiesTokenCacheHelper.ForApiCalls(context, AzureADDefaults.CookieScheme), null);
-
-            string userObjectId = user.FindFirst("http://schemas.microsoft.com/identity/claims/objectidentifier").Value;
-            string tenantId = user.FindFirst("http://schemas.microsoft.com/identity/claims/tenantid").Value;
+            CreateApplicationIfNeeded(context, null, AzureADDefaults.CookieScheme);
+            return await GetAccessTokenOnBehalfOfUser(context.User, scopes);
+        }
+
+
+        private static string GetAccountId(ClaimsPrincipal user)
```

Comment:
Extracting this method (single responsibility).
Would be actually nice to have in MLSA.NET.